classes/class-export.php

```diff
@@ -0,0 +1,242 @@
+					$user      = new Author( (int) $record->user_id, (array) maybe_unserialize( $record->user_meta ) );
+					$row_out[ $column_name ] = $user->get_display_name();
+					break;
+
+				case 'connector':
+					$row_out[ $column_name ] = $record->connector;
+					break;
+
+				case 'context':
+					$row_out[ $column_name ] = $record->context;
+					break;
+
+				case 'action':
+					$row_out[ $column_name ] = $record->{$column_name};
+					break;
+
+				case 'blog_id':
+					$row_out[ $column_name ] = $record->blog_id;
+					break;
+
+				case 'ip' :
+					$row_out[ $column_name ] = $record->{$column_name};
+					break;
+			}
+		}
+
+		return $row_out;
+	}
+
+	/**
+	 * Increase pagination limit for CSV Output
+	 *
+	 * @param int $records_per_page Old limit for records_per_page.
+	 */
+	public function disable_paginate( $records_per_page ) {
+		return 10000;
+	}
+
+	/**
+	 * Expand columns for CSV Output
+	 *
+	 * @param array $columns Columns currently registered to the list table being exported.
+	 * @return array New columns for exporting.
+	 */
+	public function expand_columns( $columns ) {
+		$new_columns = array(
+			'date'      => $columns['date'],
+			'summary'   => $columns['summary'],
+			'user_id'   => $columns['user_id'],
+			'connector' => __( 'Connector', 'stream' ),
+			'context'   => $columns['context'],
+			'action'    => $columns['action'],
+			'ip'        => $columns['ip'],
+		);
+
+		if ( is_multisite() && is_plugin_active_for_network( $this->plugin->locations['plugin'] ) ) {
+			$new_columns['blog_id'] = __( 'Blog ID', 'stream' );
+		}
+
+		return $new_columns;
+	}
+
+	/**
+	 * Registers all available exporters
+	 *
+	 * @return void
+	 */
+	public function register_exporters() {
+		$exporters = array(
+			'csv',
+			'json',
+		);
+
+		$classes = array();
+		foreach ( $exporters as $exporter ) {
+			include_once $this->plugin->locations['dir'] . '/exporters/class-exporter-' . $exporter .'.php';
+			$class_name = sprintf( '\WP_Stream\Exporter_%s', str_replace( '-', '_', $exporter ) );
+			if ( ! class_exists( $class_name ) ) {
+				continue;
+			}
+			$class = new $class_name();
+			if ( ! property_exists( $class, 'slug' ) ) {
+				continue;
+			}
+			$classes[ $class->slug ] = $class;
+		}
+
+		/**
+		 * Allows for adding additional exporters via classes that extend Exporter.
+		 *
+		 * @param array $classes An array of Exporter objects. In the format exporter_slug => Exporter_Class()
+		 */
+		$this->exporters = apply_filters( 'wp_stream_exporters', $classes );
+
+		// Ensure that all exporters extend Exporter
+		foreach ( $this->exporters as $key => $exporter ) {
+			if ( ! $this->is_valid_exporter( $exporter ) ) {
+				unset( $this->exporters[ $key ] );
+			}
+		}
+	}
+
+	/**
+	 * Checks whether an exporter class is valid
+	 *
+	 * @param Exporter $exporter The class to check.
+	 * @return bool
+	 */
+	public function is_valid_exporter( $exporter ) {
+		if ( ! is_a( $exporter, 'WP_Stream\Exporter' ) ) {
+			trigger_error(
```

Comment:
We should probably trigger this error in `register_exporters()`. You should be able to call `is_valid_exporter()` on anything and simply get true or false. Otherwise it might be disjarring, akin to raising an exception with you call `is_array( 'string' )`.